main.cpp

```diff
@@ -0,0 +1,58 @@
+//CodeU: Q1
+//
+//  Created by Julia Semavina on 17.05.2017
+//
+//  problem specification:
+//      Given two strings, write a method to decide if one is a permutation of the other
+//
+//  comments:
+//      register is ignored
+//
+
+#include <iostream>
+#include <map>
+
+
+char lowercase(char a) {
+    if (a <= 'Z' && a >= 'A')
+        return (a - ('Z' - 'z'));
+    return a;
+}
+
+bool is_permutation(std::string &A, std::string &B) {
+    // check if the lengths are equal
+    if (A.length() != B.length())
+        return false;
+    // make a dictionary from string A,
+    // keys - different elements of string A
+    // values - number of each key in string A
+    std::map<char, int> A_elements;
+    char x;
+    for (int i = 0; i < A.length(); ++i) {
+        x = lowercase(A[i]);
+        if (A_elements.find(x) == A_elements.end())
```

Comment:
The [] operator on maps it's special, it will construct and insert a new element if the key is not found, and the value will be the default value for the type (conveniently, 0 for ints), so you can actually just change lines 33-36 for:
A_elements[x]++;